core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -308,7 +311,7 @@ public static <ResponseT, ResultT> Builder<ResponseT, ResultT> builder() {
 
     public static final class Builder<ResponseT, ResultT> {
 
-        private long maximumBufferSize;
+        private Long maximumBufferSize;
```

Comment:
If it is null, does it defer to default (32MB If i remember correctly ) ?

Reply:
It's the `MultipartS3AsyncClient` will default to 32MB, but here if it is null, the constructor throws a NPE